packages/rocketchat-channel-settings/client/views/channelSettings.js

```diff
@@ -177,14 +177,20 @@ Template.channelSettingsEditing.onCreated(function() {
 				}
 				return true;
 			},
+			showPrivacySetting() {
+				return this.value.get();
+			},
 			canEdit() {
 				return (RocketChat.authz.hasAllPermission('edit-room', room._id) && !room['default']) || RocketChat.authz.hasRole(Meteor.userId(), 'admin');
 			},
 			save(value) {
 				const saveRoomSettings = () => {
 					value = value ? 'p' : 'c';
 					RocketChat.callbacks.run('roomTypeChanged', room);
-					return call('saveRoomSettings', room._id, 'roomType', value).then(() => {
+					return call('saveRoomSettings', room._id, {
+						'roomType': value,
+						'secretRoom': value === 'p' ? room.customFields.secretRoom : false
```

Comment:
The defaulting should not happen on the UI. instead, the method creating the room should handle it.